### ISSUE TITLE
Changing objects scales

### ISSUE THREAD

**Issue by ptycharly** [dmbfm/unity-ar-gps-location-issues#49]
Hi,
I continue exploring the CreatePointOfInterestTextMeshes script. 
I try to change the scale of objects through a slider. The recommendation for ARFoundation is to modify the scale of ARSessionOrigin so that the size of the objects placed in the scene "appear" modified.
When I modify the scale of ARSession, in my Unity scene window, it works. The TestMesh object changes size.  
When I import my project in my device, the change of scale reacts strangely. When it works, which is rare, it doesn't last. The TextMesh return to its original position after a few seconds. Otherwise, most of the time the rescaling causes the TextMesh to move up or down or disappear.
Do you have any ideas of how to solve this scale issue?

**Comment by dmbfm** [comment]
As far as I know, when you modify the scale of ARSessionOrigin, it has the opposite effect: increasing the scale will actually make the objects smaller and vice-versa. 

Also, it is not recommended to do that when using our asset.... because it needs the scale to be 1:1 to place the objects corretly. In this case, it is better to scale objects individually.

The ARSessionOrigin stuff is more useful when you don't want to mess with the scale of stuff like terrains or objects' physics.

**Comment by ptycharly** [comment]
Thank you for your prompt response.

You are right, changing the scale of the ARSessionOrigin has the opposite effect.

As I use the CreatePointOfInterestTextMeshes script, it happens that the distances of the objects I want to position are very far and therefore appear very small. That's why I wanted to use the ARSession's scale modification for all the elements.
Is that a way to change the TextMesh size according to the distance from the device?

**Comment by dmbfm** [comment]
One thing I have done in the past is to scale the object based on it's distance to the camera. Get the distance to the camera, `Vector3.Distance(gameObject.transform.position, Camera.main.transform.position)` and set the scale based on that (probably some constant * distance will do the trick).

**Comment by ptycharly** [comment]
Yes, i was thinking about something like that. I'm new with Unity and C# so the difficulty for me is to associate this distance to each game object in the CreatePointOfInterestTextMeshes script. I retrieved each distance with modification of the debugdistance script. Now i try to associate each distance to each gameobject. I'm still working on it!

**Comment by dmbfm** [comment]
Let me know if I can help in any way!  I think I will add a component that does that in the next release.

**Comment by dmbfm** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [https://github.com/dmbfm/unity-ar-gps-location-issues/issues/49](https://trello.com/c/ZJzVFlKI/5-https-githubcom-dmbfm-unity-ar-gps-location-issues-issues-49)
